include/nn/util/util_TypedStorage.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <nn/types.h>
+#include <type_traits>
+
+namespace nn::util {
+
+template <class T, size_t Size, size_t Alignment>
+struct TypedStorage {
+    typename std::aligned_storage<Size, Alignment>::type storage;
```

Comment:
As previously discussed, std::aligned_storage is deprecated in modern standards so we should avoid using it (even if that's what the original nn code does). `std::aligned_storage` was added in C++11 and so was `alignas`, so we might as well use the latter, which is less error prone to use and *not* deprecated:

```suggestion
    alignas(Alignment) char storage[Size];
```